### ISSUE TITLE
Question: combining twind runtime with tailwind buildtime

### ISSUE THREAD

**Issue by andrei9669** [tw-in-js/twind#78]
I found this library in my search for runtime tailwind.
My use case is basically this that I would like to have a set of default tailwind styles added to the application beforehand and then when the user wants to change the style of an item at runtime, use twind to update the existing style tag.
I would like to prevent the import of tailwind classes with twind if I already have some classes packaged in my app.

Question is, is it possible to do such a thing with this library?

**Comment by sastan** [comment]
It would be good to know how a user would change the style. You could use [tw](https://github.com/tw-in-js/twind/blob/main/docs/tw.md) or the [shim](https://github.com/tw-in-js/twind/blob/main/docs/installation.md#twindshim)

>  I would like to have a set of default tailwind styles added to the application beforehand

These would be generated by tailwindcss? Or by twind using [Static Extraction](https://github.com/tw-in-js/twind/blob/main/docs/ssr.md)?

twind creates or reuses a style tag identified by the id `__twind`. It preserves all styles within that stylesheet and appends runtime generates styles to that stylesheet. 

> I would like to prevent the import of tailwind classes with twind if I already have some classes packaged in my app.

If i understand you correctly you do not want twind to generate styles that are already available on the page. This is something we have thought a lot about but this would lead to inconsistencies as the order of tailwind css classes within the stylesheet matters. twind ensures that all css classes it inject are in the correct order. Parsing the existing style tags is quite difficult and may lead to unexpected bugs.

I hope that this helps. Please ask for more details – we like to help you.

**Comment by andrei9669** [comment]
>This is something we have thought a lot about but this would lead to inconsistencies as the order of tailwind css classes within the stylesheet matters. twind ensures that all css classes it inject are in the correct order. Parsing the existing style tags is quite difficult and may lead to unexpected bugs.

Oh, I didn't know that. Altho, I think it could be probably possible, but not easy, to somehow create our own purge cli.

>These would be generated by tailwindcss? Or by twind using Static Extraction?

Either would do, whatever works. But the app itself wouldn't be SSR.

>It would be good to know how a user would change the style. You could use tw or the shim

Well, that was the idea, I would have a separate menu where a user could change the style of their own UI with tw classes.

**Comment by sastan** [comment]
@andrei9669 Which other open points do you have? Did we address everything?

**Comment by andrei9669** [comment]
No, I got all I wanted. But one last thing, would you consider this library production ready?

**Comment by sastan** [comment]
> Would you consider this library production ready?

TLDR: Yes.

We are quite happy with the existing API and do not expect breaking changes before a 1.0 release. My company is already using it and supporting the development.

We support all tailwind classes and variants (and some more).

Before a 1.0 release we want to add some features like component composition #73, a react styled API #7 and support tailwindcss plugins #32. These may affect some internal APIs but we try to keep the public API stable.

Performance is already great and should not be an issue.

**Comment by andrei9669** [comment]
awesome, then I might think of using it as well :D

**Comment by sastan** [comment]
Great! Let us know how it goes and feel free to contact us if you hit any walls or need some features. 🤞

**Comment by andrei9669** [comment]
Will do
